src/serac/physics/solid_functional.hpp

```diff
@@ -368,17 +367,16 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
             deformation_grad = deformation_grad + du_dX;
           }
 
-          // Note that the jacobian needs the fixup for the shape displacement.
-          // The material returns Kirchoff stress, which contains det(du / dX'). We want
-          // The final Jacobian to be det(du / dX), so we compute
-          // det(du / dX) = det(du / dX') * det(dX' / dX) = det(du / dX') * det(I + dp / dX)
-          auto flux = dot(stress, transpose(inv(deformation_grad))) * det(I + dp_dX);
+          auto flux = dot(stress, transpose(inv(deformation_grad))) * det(deformation_grad);
 
           // This transpose on the stress in the following line is a
           // hack to fix a bug in the residual operator. The stress
           // should be transposed in the contraction of the Piola
           // stress with the shape function gradients.
           //
+          // Note that the mass part of the return is integrated in the perturbed reference
+          // configuration, hence the det(I + dp_dx) = det(dX'/dX)
+          //
           // TODO: fix the residual implementation and remove this transpose.
           return serac::tuple{material.density * d2u_dt2 * det(I + dp_dX), transpose(flux)};
```

Comment:
As I said in the earlier comment, I believe the PR is missing a factor of `det(I + dp_dX)` on the stress. If I'm right, maybe we should do that multiplication on this line so that there is symmetry between the D'Alembert force and the flux. Also, one comment then covers both.